### PR TITLE
[MAINT] reduce occurence of resampling related warnings

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none but relates to #4412, #4441, #4450

Changes proposed in this pull request:

- some internal occurrence of force `resample_img` and `resample_to_img` were not set with `force_resample=False` leading to extra warnings that users cannot act on despite the warning message implying that they can.
- change dtype of some of the data used to test resample_img to reduce occurences of "casting to float" warnings in tests

### CODE REVIEW

File: examples/06_manipulating_images/plot_affine_transformation.py [hunk 1]
```diff
@@ -52,18 +52,22 @@
 
 grid = np.mgrid[0:192, 0:128]
 circle = (
-    np.sum((grid - np.array([32, 32])[:, np.newaxis, np.newaxis]) ** 2, axis=0)
+    np.sum(
+        (grid - np.array([32.0, 32.0])[:, np.newaxis, np.newaxis]) ** 2, axis=0
```
Comment (Remi-Gau) [inline examples/06_manipulating_images/plot_affine_transformation.py:56]: unrelated to but removes some `cast to float` warning when running the example

**Comment by Remi-Gau** [comment]
by my count there was about 70 such warnings in the doc build and about a 100 of them in the test suite

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Review by htwangtw — APPROVED** [review]
Thanks LGTM!

**Comment by Remi-Gau** [comment]
thanks for the reviews @bthirion @htwangtw

**Pull request merged**